DataDog/opentelemetry-collector-contrib
# Create initial outline for Datadog exporter

### What does this PR do?

- Create an initial outline for a future Datadog exporter
- Add support for basic configuration settings: API key, site and custom endpoint (the latter will be mainly useful for debugging)
- Define metrics and traces exporter no-op creators: they sanitize the configuration and then fail loudly.

### How to run and test this

- Running `make` will run all linters and tests.
- To run the collector with the Datadog exporter, modify the example under the `examples` folder using the `exporter/datadogexporter/examples` syntax, then build and run it: `docker-compose build && docker-compose up`. Right now it should just log that it's checking the config and then log an error indicating neither metrics nor traces are supported.
